embed_grader/serapis/views/tasks.py

```diff
@@ -105,8 +109,46 @@ def delete_assignment_task(request, task_id):
 
     if not user.has_perm('modify_assignment', course):
         return HttpResponse("Not enough privilege")
-    
+
     task.delete()
 
     return HttpResponseRedirect(
             reverse('assignment', kwargs={'assignment_id': assignment.id}))
+
+@login_required(login_url='/login/')
+def zip_input_files(request, task_id):
+    try:
+        task = AssignmentTask.objects.get(id=task_id)
+    except AssignmentTask.DoesNotExist:
+        return HttpResponse("Assignment task cannot be found")
+
+    user = User.objects.get(username=request.user)
+    assignment = task.assignment_fk
+    course = assignment.course_fk
+    if not user.has_perm('view_assignment', course):
+        return HttpResponse("Not enough privilege")
```

Comment:
Line 128-129 actually do the permission check, so if a student does not enroll in a course, then he does not have the view_assignment permission for the course, and thus is not able download any file associated with the assignments for that course. @TimeString

Reply:
You're right!